### ISSUE TITLE
Examples crash with Style loading race condition

### ISSUE THREAD

**Issue by kmadsen** [mapbox/mapbox-navigation-android#2942]
**Android API:**
**Mapbox Navigation SDK version:**

### Steps to trigger behavior

 1. Open examples, Faster Route
 2. Crash sometimes

```
2020-05-08 14:16:51.578 9478-9478/com.mapbox.navigation.examples E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.mapbox.navigation.examples, PID: 9478
    com.mapbox.mapboxsdk.location.LocationComponentNotInitializedException: The LocationComponent has to be activated with one of the LocationComponent#activateLocationComponent overloads before any other methods are invoked.
        at com.mapbox.mapboxsdk.location.LocationComponent.checkActivationState(LocationComponent.java:1712)
        at com.mapbox.mapboxsdk.location.LocationComponent.forceLocationUpdate(LocationComponent.java:913)
        at com.mapbox.navigation.examples.core.FasterRouteActivity$MyLocationEngineCallback.onSuccess(FasterRouteActivity.kt:290)
        at com.mapbox.navigation.examples.core.FasterRouteActivity$MyLocationEngineCallback.onSuccess(FasterRouteActivity.kt:283)
        at com.mapbox.android.core.location.GoogleLocationEngineImpl$GoogleLastLocationEngineCallbackTransport.onSuccess(GoogleLocationEngineImpl.java:136)
        at com.mapbox.android.core.location.GoogleLocationEngineImpl$GoogleLastLocationEngineCallbackTransport.onSuccess(GoogleLocationEngineImpl.java:125)
        at com.google.android.gms.tasks.zzn.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

### Expected behavior

Tricky lifecycle and asynchronous issues waiting for styles to load, classic 

### Actual behavior

When you register in onStart, this will have race conditions with styles loading. The crash above is an example of one of these race conditions. But they seem to be all over.
``` kotlin
    override fun onStart() {
        super.onStart()
        mapView.onStart()
        navigationMapboxMap?.onStart()
        mapboxNavigation.registerTripSessionStateObserver(tripSessionStateObserver)
    }
```
https://github.com/mapbox/mapbox-navigation-android/blob/752a5c65436f63f585b6037a960bec7bd7a552e1/examples/src/main/java/com/mapbox/navigation/examples/core/FasterRouteActivity.kt#L100

**Comment by kmadsen** [comment]
cc: @abhishek1508 @cafesilencio @JunDai @Guardiola31337

**Comment by Guardiola31337** [comment]
That's from Maps (`MapboxMap`) though. I was able to reproduce too

```
05-08 17:38:42.180 13896-13896/com.mapbox.navigation.examples E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.mapbox.navigation.examples, PID: 13896
    com.mapbox.mapboxsdk.location.LocationComponentNotInitializedException: The LocationComponent has to be activated with one of the LocationComponent#activateLocationComponent overloads before any other methods are invoked.
        at com.mapbox.mapboxsdk.location.LocationComponent.checkActivationState(LocationComponent.java:1712)
        at com.mapbox.mapboxsdk.location.LocationComponent.forceLocationUpdate(LocationComponent.java:913)
        at com.mapbox.navigation.examples.core.FasterRouteActivity$MyLocationEngineCallback.onSuccess(FasterRouteActivity.kt:301)
        at com.mapbox.navigation.examples.core.FasterRouteActivity$MyLocationEngineCallback.onSuccess(FasterRouteActivity.kt:294)
        at com.mapbox.android.core.location.GoogleLocationEngineImpl$GoogleLocationEngineCallbackTransport.onLocationResult(GoogleLocationEngineImpl.java:118)
        at com.google.android.gms.internal.location.zzau.notifyListener(Unknown Source)
        at com.google.android.gms.common.api.internal.ListenerHolder.notifyListenerInternal(Unknown Source)
        at com.google.android.gms.common.api.internal.ListenerHolder$zaa.handleMessage(Unknown Source)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

`FasterRouteActivity.kt:301` 👇 

```kotlin
activityRef.get()?.mapboxMap?.locationComponent?.forceLocationUpdate(it)
```

cc @LukasPaczos

**Comment by JunDai** [comment]
The issue will be almost 100% reproducible if you do a fresh install.
Here is the fix #2947
